Lib/idlelib/configdialog.py

```diff
@@ -1853,6 +1853,219 @@ def save_all_changed_extensions(self):
             self.ext_userCfg.Save()
 
 
+class FontPage:
+
+    def __init__(self, parent, tab_pages, highlight_sample):
```

Comment:
Creating a default Text in create_widgets would work for setting the options (in the two pages).  However, 'parent' is not a resettable 'option' and it must be the frame the widget is to appear in.  So the Text call must be where it is.

I believe you are referencing issues in https://en.wikipedia.org/wiki/Law_of_Demeter.

A ttk notebook is initialized empty.  Tabs are added with note.add(someframe, text='tab title').  If we used that, the new classed would be Frame subclasses.  There would still be the issue that the hightlight frame must exist before the font frame is initialized.  I will try a quick experiment.